dask/dataframe/io/parquet/arrow.py

```diff
@@ -685,9 +280,16 @@ def read_metadata(
             elif not gather_statistics:
                 gather_statistics = True
 
+        # Only pass fragments to workers if necessary
+        if fragment_row_groups is None:
+            if filters or partition_info["partition_names"]:
+                fragment_row_groups = True
+            else:
+                fragment_row_groups = False
```

Comment:
Can you clarify this a bit more? 

So I understand that if there are no filters and no partition info, only the file paths are send to the workers. But below, in `_gather_metadata` (which is called above), there is also some logic related to this to extract the file paths, but not using this keyword?

Reply:
Right - The logic is pretty similar, so I understand the confusion.  In `_gather_metadata`, our goal is to collect a list of `FileFragment` objects to act as our "metadata".  However, if the user does not want to split by row-group, and does not need to gather any statistics, then we can just use a list of file paths as our metadata (and skip statistics processing altogether).

The `fragment_row_groups` logic is different, because it only applies when the metadata is still `FileFragment`-based, and we may still be splitting by row-group and/or processing statistics.  In this case, where still **do** use fragment-based metadata to construct the `parts`/`stats`, we try to avoid passing the actual fragment objects to `read_partition` through the graph.  I added a comment to explain the reasoning a bit more (related to graph size and my own performance observations).